src/languages/dotnet.md

```diff
@@ -0,0 +1,78 @@
+
+{% codesnippet "/registry/images/examples/full/dotnet.app.yaml", language="yaml" %}{% endcodesnippet %}
+
+## Building the application
+
+.NET containers do not come with a pre-defined build process that you may have seen on other container types. Instead, the process is defined by you, using the runtime and output directory variables passed to the build environment.
+
+For simple applications, we recommend using `dotnet publish --self-contained`, while specifying the target runtime and output folders via environment variables:
+
+```yaml
+hooks:
+    build: |
+        set -xe
+        dotnet publish --self-contained --runtime "$DOTNET_DEFAULT_RUNTIME" --output "$PLATFORM_OUTPUT_DIR" -p:UseRazorBuildServer=false -p:UseSharedCompilation=false
+```
+
+When using a custom build hook, make sure to copy the `-p` toggles over, or builds will take much longer. Alternatively, when making multiple builds within one build hook, you might want to leave the servers enabled and call `dotnet build-server shutdown` at the end of your build hook.
+
+## Running the application
+
+.NET applications should be started using the `web.commands.start` directive in `.platform.app.yaml`. This ensures that the command starts at the right moment and stops gracefully when a re-deployment needs to be executed. Also, should the program terminate for any reason, it will be automatically restarted. Note that the start command _must_ run in the foreground.
+
+Incoming requests are passed to the application using either a TCP (default) or UNIX socket. The application must use the [appropriate environment variable](/configuration/app/web.html#socket-family) to determine the URI to listen on. In case of a TCP socket ([recommended](https://go.microsoft.com/fwlink/?linkid=874850)), the application must listen on `http://127.0.0.1`, using port defined by the `PORT` environment variable.
+
+
+There will be an Nginx server sitting in front of your application. We recommend serving static content via Nginx, as this allows easy control of headers (including cache headers), and also has marginal performance benefits. Note that HTTPS is also terminated at the Ngnix proxy, so the `app.UseHttpsRedirection();` line (which is in `Startup.cs` by default) should be removed. To force HTTPS-only, please refer to the [routes documentation](/configuration/routes/https.html#https).
+
+The following example configures your environment to serve the static content folders commonly found in ASP.NET MVC templates using Nginx, while routing other traffic to the .NET application.
+
+```yaml
+web:
+  locations:
+    "/":
+      root: "wwwroot"
+      allow: true
+      passthru: true
+      rules:
+        # Serve these common asset types with customs cache headers.
+        \.(jpe?g|png|gif|svgz?|css|js|map|ico|bmp|eot|woff2?|otf|ttf)$:
+          allow: true
+          expires: 300s
+
+  commands:
+    start: |
+      ./WebApplication1
```

Comment:
No need to make this a multi-line block when it's a single line.

How does one control the output file name?

Reply:
Updated. From what I can tell it's taken from the name you give this file and how you define within it https://github.com/platformsh/template-builder/blob/aspnet/templates/aspnet-core/files/PlatformshAspNetCore.sln

Reply:
Neither "Web" nor "Application" appear in that file. So, um, huh?

Reply:
Oh I'm just saying that that file would be called `WebApplication1.sln` with that name throughout this line 
```
Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "PlatformshAspNetCore", "PlatformshAspNetCore.csproj", "{6150B0DB-45B5-4429-9410-2BA4E68874CC}"
```
so then that defines https://github.com/platformsh/template-builder/blob/446ccc0925747318576f765163a02b33190edf24/templates/aspnet-core/files/.platform.app.yaml#L28

Reply:
Ah, ok.